### ISSUE TITLE
Integrate poems into knowl production

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#713]
The `poem` element needs to be integrated into the whole blocks/knowl production framework.  aAt a minimum, it should change to an `article` and its CSS should no longer be hard-coded.

Similarly, the entire CSS infrastructure for a poem should not be hard-coded.

**Comment by klevasseur** [comment]
Thanks Rob.  I open most of our chapters with a limerick, so doing this would be appreciated.

Ken

**Comment by rbeezer** [comment]
It'll get done.  But likely needs some big CSS work from David.  Not hard, but 
he's tied up for a few days, and when he's back, then I'm AWOL.  ;-)

Levasseur sounds French, not Irish. ;-)

**Comment by davidfarmer** [comment]
Just post a test link, preferably containing content that also is
available with the old inline styling.  I'll get to it.

**Comment by davidfarmer** [comment]
Isn't there also some new or pending implimentation of epigraphs?

**Comment by kcrisman** [comment]
CS students studying in Lowell
Required their books to have soul.
Along came their teacher
Who asked for a feature:
A poem that lived in a knowl.

**Comment by rbeezer** [comment]
Now I have a sample to use for testing.

**Comment by rbeezer** [comment]
@davidfarmer Thanks!  I naively took out all `@style` attributes.  And also changed outermost element to an `article` with the poem's title in an `h5`.  Result at
http://mathbook.pugetsound.edu/beta/20170907-poem-css/poetry.html

You can compare with official output:
http://mathbook.pugetsound.edu/examples/humanities/html/poetry.html

Search `mathbook-html.xsl` for `match="poem"` if it is easier to see algorithmically created CSS rather than reverse-engineering bland output.

Yes, epigram is out there, and something else.  Had my head down working on numbers.

**Comment by klevasseur** [comment]
Nice, you're not Irish either!

**Comment by rbeezer** [comment]
Poems should now participate as targets of cross-references, including from the index.  At 6011c3662367bb143fb438e1d69304354fa9b6d8 and just prior.

Tested, but not super-thoroughly, so @klevasseur, holler if anything seems wrong with your limericks.  Ooops forgot to add the one by @kcrisman.  Coming up.

I had to inject some inline CSS stuff into the (new) general block and knowl production.  With #732, that should go away.  I'll locate that stuff there.
